src/components/common/CodeReview.js

```diff
@@ -351,6 +369,15 @@ class CodeReview extends Component {
     });
   }
 
+  handleHideAllComments = (checked) => {
+    const { lineCommentsHiddenState: lineCommentsHiddenStatePrevious } =
+      this.state;
+    const lineCommentsHiddenState =
+      lineCommentsHiddenStatePrevious.fill(checked);
+    // set hidden state
+    this.setState({ lineCommentsHiddenState });
```

Comment:
You should do this with a `setState` with a callback that can access its previous state.